aiohttp/web_response.py

```diff
@@ -635,3 +654,25 @@ def json_response(data=sentinel, *, text=None, body=None, status=200,
             text = dumps(data)
     return Response(text=text, body=body, status=status, reason=reason,
                     headers=headers, content_type=content_type)
+
+
+async def async_json_response(data=sentinel, *, text=None, body=None,
```

Comment:
I think we shouldn't have this function. While it may not be simple to improve `json_response`, it's much more better to explain users to do dump outside of it call and pass the result as `text` argument to original `json_response`.

Reply:
I think we should because it plays into the `_BODY_LENGTH_THREAD_CUTOFF` value, further I think `json_response` should then be deprecated

Reply:
That's all because global private constant `_BODY_LENGTH_THREAD_CUTOFF`. Once it will be configurable, composition or creating custom object instead will be the answer. Say, `JsonResponse` one which will do dump properly in executor and old `json_response` will be just a handy helper for compability reasons. Users even wouldn't need to change their code a lot to gain benefits from defaults.

Reply:
ya but remember the original reason for this helper method, it's to avoid people from having to re-implement things the same way, I think if we can provide something that helps reduce people typing the same code over and over it's a win.   With the above parameterization change I think it makes even more sense, let me know what you think.  Sorry going to keep pushing for a bit more because I know for me it reduces a bunch of our services from having to do this independently.

Reply:
removed per discussion below